crypto/secp256k1/secp256k1.go

```diff
@@ -43,11 +65,18 @@ func (privKey PrivKey) Bytes() []byte {
 // PubKey performs the point-scalar multiplication from the privKey on the
 // generator point to get the pubkey.
 func (privKey PrivKey) PubKey() crypto.PubKey {
-	_, pubkeyObject := secp256k1.PrivKeyFromBytes(privKey)
-
-	pk := pubkeyObject.SerializeCompressed()
+	privateObject, err := ethCrypto.ToECDSA(privKey)
+	if err != nil {
+		panic(err)
+	}
 
+	pk := ethCrypto.FromECDSAPub(&privateObject.PublicKey)
 	return PubKey(pk)
+
+	// _, pubkeyObject := secp256k1.PrivKeyFromBytes(privKey)
```

Comment:
I remember a convo with Sergio about this.
Final decision? Do we keep it as a comment or are we refactoring it?

Reply:
IIRC, the convo was about `genPrivKey` below. The original peppermint fork wasn't keeping the `for` loop, and we decided to keep it this time, cause it makes the code more resilient to glitches in the random source. So the status of `genPrivKey` in this PR reflects what we decided.

About the commented code in this file (and related `_test.go` file), this was done by the original fork (i.e., the original fork didn't _remove_ the code it was changing, but commented it out). When I did the forward-port, as I was doing it mechanically, I kept the changes as they were in the original fork. Feel free to remove them (I would!).

Reply:
Resolved (AFAIAC)